django/db/models/base.py

```diff
@@ -1421,6 +1422,26 @@ def _check_model_name_db_lookup_clashes(cls):
             )
         return errors
 
+    @classmethod
+    def _check_property_name_foreign_key_id_accessor_clashes(cls):
+        errors = []
+        property_names = cls._meta._property_names
+        foreign_key_id_accessors = (
+            '%s_%s' % (f.name, 'id') for f in cls._meta._get_fields(reverse=False)
+            if f.is_relation and f.related_model is not None
+        )
+        for accessor in foreign_key_id_accessors:
+            if accessor in property_names:
+                errors.append(
+                    checks.Error(
+                        "The property name '%s' clashes with the foreign key "
```

Comment:
"foreign key id" may not be the best wording. For example, this also applies to subclasses like OneToOneField.